src/Util/Autoload/ClassLoader.php

```diff
@@ -49,10 +49,10 @@ public function addClass(ReflectionClass $reflectionClass): void
     /**
      * @throws FailedToLoadClass
      */
-    public function __invoke(string $classToLoad): bool
+    public function __invoke(string $classToLoad): void
```

Comment:
Wait, why did we change this signature?

Isn't a `void` function requirement (as parameter) compatible with a `bool` implementation anyway?

Reply:
There's bug reported by PHPStan: `Parameter #1 $callback of function spl_autoload_register expects (callable(string): void)|null, $this(Roave\BetterReflection\Util\Autoload\ClassLoader) given`

Reply:
I think it needs to be baselined: any `callable():void` parameter requirement is compatible with `callable():mixed`.

https://psalm.dev/r/c0bcfcb184

Reply:
I remember this has been a big discussion on the internals. Right now the PHP covariance rules do not allow going from `void` to `bool`: https://3v4l.org/PD4Zt

I understand both points of view - the child class can return something without breaking the contract. At the same time, not allowing `callable():mixed` to be passed to `callable():void` might find some bugs. You might have thought you need to return something but actually you don't - which is a useful thing to report by SA.

Reply:
@Ocramius Should I add to to the baseline or are you ok with the comment by @ondrejmirtes ? :)

Reply:
TBH disagree with @ondrejmirtes here, heh 😬

Reply:
https://github.com/Roave/BetterReflection/pull/750/commits/6129a2ed54183b08f062e39442128bab6d0098c9